### DISCUSSION TITLE
rg in directory with lots of unpinned files

### DISCUSSION THREAD

**Discussion by jessekv** [BurntSushi/ripgrep#1620] [General] [answered]
I typed `rg` in the root of a directory that uses the Windows 10 API for offline files. My computer locked up and crashed shortly after, probably because most of the files were not `Pinned` (they also have the attribute `RecallOnDataAccess`) and ripgrep likely triggered the download of thousands of files. Is there a way to scan offline-only files with ripgrep?

**Comment by BurntSushi** [comment, marked as answer]
I don't see how, no. ripgrep doesn't know about remote file systems. That's all transparent to ripgrep. If you tell it to search a directory then it will. As far as I can tell, the only way to limit it to offline files is to get a list of offline files and feed that list as arguments to ripgrep.
